pkg/engine/executor/parse_logfmt.go

```diff
@@ -1,109 +1,15 @@
 package executor
 
 import (
-	"sort"
-	"unsafe"
-
 	"github.com/apache/arrow-go/v18/arrow"
 	"github.com/apache/arrow-go/v18/arrow/array"
 	"github.com/apache/arrow-go/v18/arrow/memory"
-
 	"github.com/grafana/loki/v3/pkg/engine/internal/types"
 	"github.com/grafana/loki/v3/pkg/logql/log/logfmt"
 )
 
-// BuildLogfmtColumns builds Arrow columns from logfmt input lines
-// Returns the column headers, the Arrow columns, and any error
 func BuildLogfmtColumns(input *array.String, requestedKeys []string, allocator memory.Allocator) ([]string, []arrow.Array) {
-	columnBuilders := make(map[string]*array.StringBuilder)
-	columnOrder := parseKeys(input, requestedKeys, columnBuilders, allocator)
-
-	// Build final arrays
-	columns := make([]arrow.Array, 0, len(columnOrder))
-	headers := make([]string, 0, len(columnOrder))
-
-	for _, key := range columnOrder {
-		builder := columnBuilders[key]
-		columns = append(columns, builder.NewArray())
-		headers = append(headers, key)
-		builder.Release()
-	}
-
-	return headers, columns
-}
-
-// parseKeys discovers columns dynamically as lines are parsed
-func parseKeys(input *array.String, requestedKeys []string, columnBuilders map[string]*array.StringBuilder, allocator memory.Allocator) []string {
```

Comment:
`parseKeys` was move to parse.go, and abstracted to work with both logfmt and json.